src/align/operations.jl

```diff
@@ -0,0 +1,100 @@
+convert{T<:Unsigned}(::Type{T}, op::Operation) = convert(T, convert(Uint8, op))
+
+
+# Operation encoding definitions
+# ------------------------------
+
+#= Meanings of the Operations
+-----------------------------
+
++--------------+----------------+--------------------------------------------+
+| Operation    | CIGAR Rep      | Desription                                 |
++--------------+----------------+--------------------------------------------+
+| OP_GAP       | Not Applicale. | Denotes a plain stretch of gap characters. |
+| OP_MM        | CIGAR 'M'      | Match/Mismatch.                            |
+| OP_N         | CIGAR 'N'      |                                            |
+| OP_MATCH     | CIGAR '='      | Match.                                     |
+| OP_MISMATCH  | CIGAR 'X'      | Mismatch.                                  |
+| OP_SCLIP     | CIGAR 'S'      | Soft clipping.                             |
+| OP_HCLIP     | CIGAR 'H'      | Hard clipping.                             |
+| OP_INSERT    | CIGAR 'I'      | Insertion.                                 |
+| OP_DELETE    | CIGAR 'D'      | Deletion.                                  |
+| OP_PAD       | CIGAR 'P'      | Padding.                                   |
+| OP_INVALID   | Invalid        | Invalid operation.                         |
++--------------+----------------+--------------------------------------------+
+
+=#
+
+for op = [
+  (:OP_GAP, 0), (:OP_MM, 1), (:OP_N, 2), (:OP_MATCH, 3),
+  (:OP_MISMATCH, 4), (:OP_SCLIP, 5), (:OP_HCLIP, 6), (:OP_INSERT, 7),
+  (:OP_DELETE, 8), (:OP_PAD, 9), (:OP_INVALID, 255)
+  ]
+
+  @eval const $(op[1]) = convert(Operation, Uint8($(op[2])))
+
+end
+
+
+# Conversion from characters to operations
+# ----------------------------------------
+
+# Lookup table for conversion from Char to Operation
+const char_to_op = [
+  OP_GAP, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_INVALID, OP_MATCH, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_DELETE, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_HCLIP, OP_INSERT, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_MM, OP_N, OP_INVALID,
+  OP_PAD, OP_INVALID, OP_INVALID, OP_SCLIP, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_MISMATCH, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_DELETE, OP_INVALID, OP_INVALID, OP_INVALID, OP_HCLIP,
+  OP_INSERT, OP_INVALID, OP_INVALID, OP_INVALID, OP_MM,
+  OP_N, OP_INVALID, OP_PAD, OP_INVALID, OP_INVALID,
+  OP_SCLIP, OP_INVALID, OP_INVALID, OP_INVALID, OP_INVALID,
+  OP_MISMATCH
+]
+
+
+function convert(::Type{Operation}, c::Char)
+  @inbounds op = '-' <= c <= 'x' ? char_to_op[c - '-' + 1] : OP_INVALID
+  @assert op != OP_INVALID error("$(c) is not a valid alignment operation.")
```

Comment:
This raises an assertion error, that can be confusing to users. What about raising an `error` inside of an `if` clause?